plugins/CoreVisualizations/javascripts/seriesPicker.js

```diff
@@ -80,40 +70,62 @@
 
             var self = this;
 
+            var selectedColumns = this.selectableColumns
+                .filter(isItemDisplayed)
+                .map(function (columnConfig) {
+                    return columnConfig.column;
+                });
+
+            var selectedRows = this.selectableRows
+                .filter(isItemDisplayed)
+                .map(function (rowConfig) {
+                    return rowConfig.matcher;
+                });
+
             // initialize dom element
-            this.domElem = $(doc.createElement('a'))
-                .addClass('jqplot-seriespicker')
-                .attr('href', '#')
-                .html('+')
+            var seriesPicker = '<piwik-series-picker'
+                + ' multiselect="' + this.multiSelect + '"'
```

Comment:
should we call some escaping here? I see it seems to hold a boolean value so should be fine just to prevent possible XSS it in the future (or maybe something like `this.multiselect ? 'true' : false'`)? Also wondering: can we use old `<div piwik-series-picker` syntax with components? I checked at it seems to be still working with IE10 so reckon no other browsers will have trouble with this but not sure. I presume custom element names are supported everywhere else?

Reply:
Good catch, completely missed it.

> Also wondering: can we use old `<div piwik-series-picker` syntax with components?

I think angularjs 1.6 does not formally allow this (though oddly, angular 2+ supports it). See the 'restrict' part of https://docs.angularjs.org/guide/component#comparison-between-directive-definition-and-component-definition .

From some googling it might be allowed, but not intentionally.

> I presume custom element names are supported everywhere else?

I can't find a proper browser support page, they all show support for the whole custom element feature, not just unknown tags. From what I remember only IE and some older versions of Firefox had this problem. Not sure what version of IE, but since angularjs 1.6 doesn't really allow attribute names for components, I'm assuming it works for whatever browser versions angular 1.6 supports.